internal/controller/connectionsecret/connectionsecrets.go

```diff
@@ -17,198 +17,266 @@ package connectionsecret
+		if err != nil {
+			return "", err
+		}
+		atlasProject, err := r.ResolveProject(ctx, sdkClientSet.SdkClient20250312002, pair.User)
+		if err != nil {
+			return "", err
+		}
+		if atlasProject.Name != "" {
+			return atlasProject.Name, nil
 		}
-		secrets = append(secrets, secretName)
-		ctx.Log.Debugw("Ensured connection Secret up-to-date", "secretname", secretName)
 	}
 
-	if len(secrets) > 0 {
-		recorder.Eventf(&dbUser, "Normal", ConnectionSecretsEnsuredEvent, "Connection Secrets were created/updated: %s", strings.Join(secrets, ", "))
+	return "", fmt.Errorf("unable to resolve ProjectName")
+}
+
+// handleDelete manages the case where we will delete the connection secret
+func (r *ConnectionSecretReconciler) handleDelete(
+	ctx context.Context, req ctrl.Request, ids ConnSecretIdentifiers, pair *ConnSecretPair) (ctrl.Result, error) {
+	log := r.Log.With("ns", req.Namespace, "name", req.Name)
+
+	// ProjectName is required for ConnectionSecret metadata.name to delete
+	projectName, err := r.resolveProjectName(ctx, ids, pair)
+	if projectName == "" {
+		err = fmt.Errorf("project name is empty")
+	}
+	if err != nil {
+		log.Errorw("failed to resolve project name", "reason", workflow.ConnSecretUnresolvedProjectName, "error", err)
+		return workflow.Terminate(workflow.ConnSecretUnresolvedProjectName, err).ReconcileResult()
 	}
 
-	if err := cleanupStaleSecrets(ctx, k8sClient, project.ID, dbUser); err != nil {
-		return workflow.Terminate(workflow.DatabaseUserStaleConnectionSecrets, err)
+	log.Debugw("project name resolved for delete")
+
+	name := CreateK8sFormat(projectName, ids.ClusterName, ids.DatabaseUsername)
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: req.Namespace,
+		},
 	}
 
-	if requeue {
-		return workflow.InProgress(workflow.DatabaseUserConnectionSecretsNotCreated, "Waiting for deployments to get created/updated")
+	// Delete the secret
+	if err := r.Client.Delete(ctx, secret); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Debugw("no secret to delete; already gone")
+			return workflow.TerminateSilently(nil).WithoutRetry().ReconcileResult()
+		}
+		log.Errorw("unable to delete secret", "reason", workflow.ConnSecretFailedDeletion, "error", err)
+		return workflow.Terminate(workflow.ConnSecretFailedDeletion, err).ReconcileResult()
 	}
-	return workflow.OK()
+
+	log.Infow("secret deleted", "reason", workflow.ConnSecretDeleted)
+	r.EventRecorder.Event(secret, corev1.EventTypeNormal, "Deleted", "ConnectionSecret deleted")
+	return workflow.TerminateSilently(nil).WithoutRetry().ReconcileResult()
 }
 
-func cleanupStaleSecrets(ctx *workflow.Context, k8sClient client.Client, projectID string, user akov2.AtlasDatabaseUser) error {
-	if err := removeStaleByScope(ctx, k8sClient, projectID, user); err != nil {
-		return err
+// handleUpdate manages the case where we will create or update the connection secret
+func (r *ConnectionSecretReconciler) handleUpdate(
```

Comment:
If it creates or updates, let have a better name.
```suggestion
func (r *ConnectionSecretReconciler) handleUpsert(
```